index.js

```diff
@@ -256,6 +258,9 @@ const ContainerService = {
 
     webExtension.startup().then(api => {
       api.browser.runtime.onMessage.addListener((message, sender, sendReply) => {
+        if (message.method === "checkForMajorUpgrade") {
+          sendReply(reason === "upgrade" && MAJOR_VERSIONS.indexOf(self.version) > -1);
```

Comment:
I meant move this code into WebExtension land. As mentioned the `runtime.onInstalled` may not fire but I don't think we need that to know if the user has seen it. We can store `latestSeenNewIndicator` or all of them etc. When the user pops open the doorhanger we can dismiss the badge indicator and store that they saw it.